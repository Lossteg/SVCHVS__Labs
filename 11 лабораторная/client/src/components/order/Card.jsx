import React from "react";
import {
  Card as MuiCard,
  CardContent,
  Typography,
  Chip,
  Paper,
} from "@mui/material";

const Card = ({ order }) => {
  const { service, properties, quantity, totalPrice } = order;

  return (
    <MuiCard
      sx={{
        maxWidth: 400,
        margin: "10px",
        padding: "10px",
        display: "flex",
        flexDirection: "column",
      }}
    >
      <CardContent>
        <Typography variant="h6">{service} Order</Typography>
        <Paper
          sx={{
            display: "flex",
            flexDirection: "row",
            padding: "10px",
            justifyContent: "space-between",
          }}
        >
          <div>
            <Typography variant="subtitle1">Properties:</Typography>
            {Object.entries(properties).map(([key, value]) => (
              <Chip
                key={key}
                label={`${key}: ${value}`}
                size="small"
                sx={{ margin: "2px" }}
              />
            ))}
          </div>
          <div>
            <Typography variant="subtitle1">Quantity:</Typography>
            <Typography variant="body1">{quantity}</Typography>
          </div>
          <div>
            <Typography variant="subtitle1">Total Price:</Typography>
            <Typography variant="body1">${totalPrice}</Typography>
          </div>
        </Paper>
      </CardContent>
    </MuiCard>
  );
};

export default Card;

