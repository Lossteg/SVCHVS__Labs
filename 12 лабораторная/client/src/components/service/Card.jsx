import React from "react";
import {
  Card as MuiCard,
  CardContent,
  CardActions,
  Button,
  Typography,
} from "@mui/material";

const Card = ({  name, imageUrl, onDetailsClick, onMakeOrderClick }) => {
  return (
    <MuiCard
      sx={{
        maxWidth: 300,
        margin: "0 5 3 0",
        alignSelf: "center",
        padding: "2px",
      }}
    >
      <img
        src={imageUrl}
        alt={name}
        style={{ width: "300px", height: "280px" }}
      />
      <CardContent>
        <Typography variant="h6">{name}</Typography>
      </CardContent>
      <CardActions style={{ display: "flex", justifyContent: "space-between" }}>
        <Button size="small" onClick={onDetailsClick}>
          Read More
        </Button>
          <Button size="small" color="primary" variant="contained" onClick={onMakeOrderClick}>
            Order
          </Button>
      </CardActions>
    </MuiCard>
  );
};

export default Card;

