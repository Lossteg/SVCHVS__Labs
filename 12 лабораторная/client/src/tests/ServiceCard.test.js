import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";
import Card from '../components/service/Card';

describe('Card component', () => {
  const mockProps = {
    name: 'Test Card',
    imageUrl: 'test-image.jpg',
    onDetailsClick: jest.fn(),
    onMakeOrderClick: jest.fn(),
  };

  it('renders the Card component with correct content', () => {
    const { getByText, getByAltText } = render(<Card {...mockProps} />);

    expect(getByText(mockProps.name)).toBeInTheDocument();

    expect(getByAltText(mockProps.name)).toBeInTheDocument();
    expect(getByAltText(mockProps.name).getAttribute('src')).toBe(mockProps.imageUrl);

    expect(getByText('Read More')).toBeInTheDocument();
    expect(getByText('Order')).toBeInTheDocument();
  });

  it('calls onDetailsClick when "Read More" button is clicked', () => {
    const { getByText } = render(<Card {...mockProps} />);
    const readMoreButton = getByText('Read More');

    fireEvent.click(readMoreButton);

    expect(mockProps.onDetailsClick).toHaveBeenCalled();
  });

  it('calls onMakeOrderClick when "Order" button is clicked', () => {
    const { getByText } = render(<Card {...mockProps} />);
    const orderButton = getByText('Order');

    fireEvent.click(orderButton);

    expect(mockProps.onMakeOrderClick).toHaveBeenCalled();
  });
  it('ServiceCard component matches snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Card {...mockProps} />
        </MemoryRouter>
      )
      .toJSON();
  
    expect(tree).toMatchSnapshot();
  });
});
