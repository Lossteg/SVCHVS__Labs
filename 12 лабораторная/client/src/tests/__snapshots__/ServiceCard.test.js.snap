// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component ServiceCard component matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1w5642c-MuiPaper-root-MuiCard-root"
>
  <img
    alt="Test Card"
    src="test-image.jpg"
    style={
      Object {
        "height": "280px",
        "width": "300px",
      }
    }
  />
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Test Card
    </h6>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing css-dnrpxu-MuiCardActions-root"
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1knaqv7-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Read More
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Order
    </button>
  </div>
</div>
`;
